portindex2json/portindex2json.tcl

```diff
@@ -64,7 +64,58 @@ proc remove_extra_braces {text} {
     return [regsub -all {[\{\}]} $text ""]
 }
 
-set fd [open [lindex $argv 0] r]
+proc print_usage {} {
+    puts stdout "\nUsage: /path/to/tclsh portindex2json.tcl <path-to-portindex> \[-info key=vlaue\]"
+    puts stdout "\nEach '-info key=value' combinaton adds one key-value pair to the JSON output inside the parent key: \"info\""
+    puts stdout "\nYou may provide any number of '-info key=value' combinations."
+    puts stdout "\nThe arguments <path-to-portindex> and `-info key=value` are position independent.\n"
+}
+
+set filecount 0
+
+for {set i 0} {$i < $argc} {incr i} {
+    set arg [lindex $argv $i]
+    switch -regex -- $arg {
+        {^-.+} {
+            if {$arg eq "-info"} {
+                incr i
+                set argument [split [lindex $argv $i] =]
+                if {[lindex $argument 1] eq ""} {
+                    puts stderr "\nError: No value provided for the key: [lindex $argument 0]. The correct syntaxt is -info key=value. Use -help for more\n"
+                    exit 1
+                } else {
+                    set json_info([lindex $argument 0]) [::json::write string [lindex $argument 1]]
+                }
+            } elseif {$arg eq "-help" || $arg eq "-h" || $arg eq "--help" } {
+                print_usage
+                exit 0
+            } else {
+                puts stderr "\nError: $arg is an invalid argument. Expecting -info, -help.\n"
+                exit 1
+            }
+        }
+        default {
+            incr filecount
+            set portindex_file [lindex $argv $i]
+        }
+    }
+}
+
+if {$filecount eq 1} {
+    if { [catch {open $portindex_file r} fid] } {
+        puts stderr "\nError: Could not open $portindex_file. Make sure the path to the PortIndex file is correct\n"
+        exit 1
+    } else {
+        set fd [open $portindex_file r]
+    }
+} elseif {$filecount eq 0} {
+    puts stderr "\nError: Path to PortIndex file not provided. Use -help.\n"
```

Comment:
If you do the above mentioned steps, you'll only be left with "if filename wasn't defined" case.

In that case, keep `Path to PortIndex file was not provided`, do `print_usage` and exit.

Reply:
I have tried to do the changes. Could you please take a look again? Thank you